mitodl/mitxonline
# 3649 all mitxonline courses should have a department

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/3649

### Description
Makes "department" a required field for courses and programs.  Updates the create_courseware management command to require department.

This PR removes functionality that created courses or programs in some management commands where the main functionality was not intended to create courses or programs.  This should simplify those management commands and keep them specific in their intended use cases.

### How can this be tested?
You must have edX (Tutor) up and running locally in order to fully test these changes.  You only need to configure Tutor so that the service user is setup in MITx Online.  This PR includes changes to update the Tutor instructions.

1. Login to Django Admin in MITx Online.
2. http://mitxonline.odl.local:8013/admin/courses/course/ Attempt to create and save a course without a department selected.  Verify an error message is shown and the course is not created.
3. http://mitxonline.odl.local:8013/admin/courses/program/ Attempt to create and save a program without a department selected.  Verify an error message is shown and the program is not created.
4. Run the `create_courseware` management command to create a course without specifying the `--department` argument.  Verify that no course is created and the command outputs an error message.
5. Run the `create_courseware` management command to create a program without specifying the `--department` argument.  Verify that no program is created and the command outputs an error message.
6. Run the `configure_instance` command and verify, via Django Admin, that courses and a program are created in MITx Online.
7. Run the `import_courserun` command.  I used the following command which references a course I created in my local instance of edX: `docker-compose run --rm web ./manage.py import_courserun --courserun course-v1:MIT+101+2022`.  Verify that you receive an error message indicating that a department must be defined.
8. Append `--dept Science` to the command in step 7 and run the command again.  Verify, via Django Admin in MITx Online, that the course and corresponding course_run were successfully created.
